openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -234,9 +237,23 @@ def get_cohort(user, course_key, assign=True, use_cached=False):
     # Otherwise assign the user a cohort.
     try:
         with transaction.atomic():
+            # If learner has been pre-registered in a cohort, get that cohort. Otherwise assign to a random cohort.
+            course_user_group = None
+            for assignment in UnregisteredLearnerCohortAssignments.objects.filter(email=user.email):
```

Comment:
We will be able to filter by course_id/email combination once the table has that information. That will give us only a single result due to the uniqueness constraint.